Add 7.16 : prefer-rest-params (#2508)
airbnb/javascript: README.md

    );
    ```

>   <a name="functions--prefer-rest-params"></a><a name="7.16"></a>

Reviewer: we don't add new sections with numbered links; those are legacy.

Author: I tried to be consistent with the above points as they had been written. If still their is something wrong then can you help me how can I fixed that.

Reviewer: There’s no consistency here - just that anything added in the last couple years will not have a numbered link (`<a name`), and all the older sections will. New ones must not have it.

Author: their is any refence from where I can take a look. Then I will update this.

Reviewer: There are many other sections that are newly added; off the top of my head i can’t recall any specifically.

Author: done.

>   - [7.16](#functions--prefer-rest-params) Prefer the use `...rest` operator instead of `arguments` keyword for taking n number of arguments in function. eslint: [`prefer-rest-params`](https://eslint.org/docs/rules/prefer-rest-params)

Reviewer: rest syntax isn't an operator
```suggestion
  - [7.16](#functions--prefer-rest-params) Prefer the use `...rest` syntax instead of the `arguments` keyword for taking a variable number of arguments in a function. eslint: [`prefer-rest-params`](https://eslint.org/docs/rules/prefer-rest-params)
```

Author: done

>     > Why? It’s cleaner and descriptive, you get the array of arguments which are passed to the function.

Reviewer: ```suggestion
    > Why? It’s cleaner and descriptive, you get the array of arguments which are passed to the function. It's also the only option in an arrow function.
```

Author: done

    ```javascript
    // bad
    function foo() {
      console.log(arguments);
    }

>     // good
>     function foo(...args) {
>       console.log(args);
>     }

Reviewer: ```suggestion
    // good
    const foo = (...args) => {
      console.log(args);
    }
    
    // good
    function foo(...args) {
      console.log(args);
    }z
```

Author: done

Author: done
